automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/azure/AzureAutopauseSpec.scala

```diff
@@ -0,0 +1,128 @@
+    with TableDrivenPropertyChecks
+    with Retries
+    with CleanUp {
+
+  implicit val accessToken = Hermione.authToken()
+  "azure runtime autopauses" taggedAs ExcludeFromJenkins in { workspaceDetails =>
+    val workspaceId = workspaceDetails.workspace.workspaceId
+
+    val labelMap: java.util.HashMap[String, String] = new java.util.HashMap[String, String]()
+    labelMap.put("automation", "true")
+
+    val runtimeName = randomClusterName
+    val res =
+      for {
+        _ <- loggerIO.info(s"AzureAutoPauseSpec: About to create runtime")
+        runtimeClient <- GeneratedLeonardoClient.generateRuntimesApi
+        diskClient <- GeneratedLeonardoClient.generateDisksApi
+
+        createReq = new CreateAzureRuntimeRequest()
+          .labels(labelMap)
+          .machineSize("Standard_DS1_v2")
+          .disk(
+            new AzureDiskConfig()
+              .name(generateAzureDiskName())
+              .size(50)
+              .labels(labelMap)
+          )
+          .autoPauseThreshold(15)
+
+        _ <- IO(runtimeClient.createAzureRuntime(workspaceId, runtimeName.asString, false, createReq))
+        _ <- loggerIO.info(s"AzureAutoPauseSpec: Create runtime request submitted. Starting to poll GET")
+
+        // ------------------ Waiting for Create ------------------ //
+
+        // Verify the initial getRuntime call
+        _ <- IO.sleep(5 seconds)
+        callGetRuntime = IO(runtimeClient.getAzureRuntime(workspaceId, runtimeName.asString))
+
+        intitialGetRuntimeResponse <- callGetRuntime
+        _ <- loggerIO.info(s"initial get runtime response $intitialGetRuntimeResponse")
+        _ = intitialGetRuntimeResponse.getStatus shouldBe ClusterStatus.CREATING
+
+        _ <- loggerIO.info(
+          s"AzureAutoPauseSpec: runtime $workspaceId/${runtimeName.asString} in creating status detected"
+        )
+
+        // Verify the runtime eventually becomes Running (in 40 minutes)
+        monitorCreateResult <- streamUntilDoneOrTimeout(
+          callGetRuntime,
+          240,
+          10 seconds,
+          s"AzureAutoPauseSpec: runtime $workspaceId/${runtimeName.asString} did not finish creating after 40 minutes"
+        )(implicitly, GeneratedLeonardoClient.runtimeInStateOrError(ClusterStatus.RUNNING))
+
+        _ <- loggerIO.info(
+          s"AzureAutoPauseSpec: runtime ${workspaceId}/${runtimeName.asString} create monitor result: $monitorCreateResult"
+        )
+        _ = monitorCreateResult.getStatus() shouldBe ClusterStatus.RUNNING
+
+        // ------------------ Waiting for Autopause ------------------ //
+
+        // new IO for new stream
+        callGetRuntimeStopping = IO(runtimeClient.getAzureRuntime(workspaceId, runtimeName.asString))
+
+        _ <- loggerIO.info(
+          s"AzureAutoPauseSpec: runtime $workspaceId/${runtimeName.asString} waiting to autopause"
+        )
+
+        monitorAutoPauseResult <- streamUntilDoneOrTimeout(
+          callGetRuntimeStopping,
+          240,
```

Comment:
Can this be lower? This doesn't depend on WSM timeouts like the other polling steps, so we should be able to use something derived from the specified autopause threshold + leo's monitoring interval?

Reply:
yeah I can shorten this, I had it that high while testing